Network: v2 protocol unit test sometimes fails on CircleCI

```
time="2022-03-14T14:58:23Z" level=info msg="Discovered Nuts node (address=peer:5555), published by did:nuts:peer" module=Network
FAIL
coverage: 74.4% of statements
FAIL	github.com/nuts-foundation/nuts-node/network	22.650s
panic: sync: negative WaitGroup counter

goroutine 46 [running]:
sync.(*WaitGroup).Add(0xc00030cd14, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:74 +0x278
sync.(*WaitGroup).Done(...)
	/usr/local/go/src/sync/waitgroup.go:99
github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.TestManager_PeerDisconnected.func3.1({0xc00030b2c0, 0x47cd75}, {0x410820, 0x7f6fc3780008, 0xc00030b2d8})
	/home/circleci/project/network/transport/v2/gossip/manager_test.go:107 +0x91
github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.callSenders.func1()
	/home/circleci/project/network/transport/v2/gossip/manager.go:134 +0x130
github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.(*peerQueue).do(0xc00030b2c0, 0xc00037d6d8)
	/home/circleci/project/network/transport/v2/gossip/queue.go:70 +0xa7
github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.callSenders({0xd86acd, 0x1}, 0xc00030b2c0, {0xc00031c060, 0x1, 0x1})
	/home/circleci/project/network/transport/v2/gossip/manager.go:129 +0xcf
github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.(*manager).PeerConnected.func1()
	/home/circleci/project/network/transport/v2/gossip/manager.go:122 +0xf7
created by github.com/nuts-foundation/nuts-node/network/transport/v2/gossip.(*manager).PeerConnected
	/home/circleci/project/network/transport/v2/gossip/manager.go:113 +0x559
FAIL	github.com/nuts-foundation/nuts-node/network/transport/v2/gossip	0.294s
```

--- a/network/transport/v2/gossip/manager_test.go
+++ b/network/transport/v2/gossip/manager_test.go
@@ -21,6 +21,7 @@ package gossip
 
 import (
 	"context"
+	"runtime"
 	"sync"
 	"testing"
 	"time"
@@ -93,35 +94,23 @@ func TestManager_PeerDisconnected(t *testing.T) {
 		gMan := giveMeAgMan(t)
 		peer := transport.Peer{ID: "1"}
 		gMan.peers["2"] = gMan.peers["1"]
-		mainGroup := sync.WaitGroup{}
-		mainGroup.Add(1)
-		tickerGroupStart := sync.WaitGroup{}
-		tickerGroupStart.Add(1)
-		tickerGroupEnd := sync.WaitGroup{}
-		tickerGroupEnd.Add(1)
-		count := 0
+		currentRoutines := runtime.NumGoroutine()
+		wg := sync.WaitGroup{}
+		wg.Add(1)
 		gMan.RegisterSender(func(id transport.PeerID, refs []hash.SHA256Hash) bool {
-			count++
-			tickerGroupStart.Wait()
-			// if the ticker doesn't unregister, this one will blow up
-			mainGroup.Done()
-			tickerGroupEnd.Wait()
+			wg.Done()
 			return true
 		})
 		gMan.PeerConnected(peer)
+		assert.Equal(t, currentRoutines+1, runtime.NumGoroutine())
+		wg.Wait()
 
-		tickerGroupStart.Done()
-		mainGroup.Wait()
 		gMan.PeerDisconnected(peer)
 		gMan.peers["1"] = gMan.peers["2"] // reset administration to bypass administration test
-		tickerGroupEnd.Done()
-
-		time.Sleep(5 * time.Millisecond)
 
 		test.WaitFor(t, func() (bool, error) {
-			return count == 1, nil
-		}, 50*time.Millisecond, "timeout while waiting for mutexes")
-		assert.Equal(t, 1, count)
+			return runtime.NumGoroutine() == currentRoutines, nil
+		}, 5*time.Second, "timeout while waiting for go routine to exit")
 	})
 }
 

